test/unit/server-middleware.js

```diff
@@ -165,33 +165,9 @@ describe('Server Middleware', () => {
     })
   })
 
-  describe('for /auth/tokens', () => {
-    beforeEach(() => {
-      req.url = '/auth/tokens'
-      Handler.prototype.isRoute = jest.fn(route => route === 'tokens')
-    })
-
-    it('does not call next', async () => {
-      await middleware(req, res, next)
-      expect(next).not.toHaveBeenCalled()
-    })
-
-    it('refreshes the token', async () => {
-      Handler.prototype.setTokens.mockReturnValueOnce(true)
-
-      await middleware(req, res, next)
-
-      expect(Handler.prototype.setTokens).toHaveBeenCalled()
-    })
-
-    it('returns 401 on invalid session', async () => {
-      Handler.prototype.setTokens.mockReturnValueOnce(null)
-
-      await middleware(req, res, next)
+  it('adds a setTokens method to the request', async () => {
+    await middleware(req, res, next)
 
-      expect(Handler.prototype.setTokens).toHaveBeenCalled()
-      expect(res.writeHead).toHaveBeenCalledWith(401, { 'Content-Type': 'application/json' })
-      expect(res.end).toHaveBeenCalledWith(JSON.stringify({ error: INVALID_SESSION }))
-    })
+    expect(req.oauth).toHaveProperty('setTokens', expect.any(Function))
```

Comment:
nit: to be sure, maybe you could spy on the actual saveTokens method and check calling this calls that